### DISCUSSION TITLE
please fix 2 colors in offline vector maps

### DISCUSSION THREAD

**Discussion by moonraver** [osmandapp/OsmAnd#17857] [Maps Related]
I would like to ask you if you could fix 2 colors in Osmand maps:

- "smoothnes=bad" is almost the same color as "smoothness=very bad". There is even a big problem to see the difference when highways are next to each other. I suggest to make "smoothness=bad" more pale pink because difference between "bad" (normal car) and "very bad" (SUV) is big.

- "surface=pebblestone" has the same color as wood and concrete. I suggest to be the same color as gravel = grey. Pebblestone is the same as gravel, difference is only in the shape of stones.

Thank you

**Comment by DmitryAlexei** [comment]
Have you tried different *Map styles*? (https://osmand.net/docs/user/map/vector-maps#default-map-styles) 

Difference between smoothness=bad and smoothness=very_bad is good seen in the Offroad map style (https://osmand.net/docs/user/map/vector-maps#offroad).

You can also create your own map style (see https://osmand.net/docs/user/map/vector-maps#custom-map-style-own-map-style) or propose a PR in default rendering.xml (https://github.com/osmandapp/OsmAnd-resources/blob/master/rendering_styles/default.render.xml) for a common use.

**Comment by moonraver** [comment]
I edited following line in my phone app in file "default.renderer.xml":

<renderingConstant name="surfaceGradeBadColor" value="#febcbc"/>

I changed "value=#febcbc" to a different hex color "value=#ddfebc" but nothing happened with the color in Osmand app. So I tried to force stop app and clear app cache but it didn't help. Is there something I should do to change the color?
I use Android version of Osmand 4.6 latest build released every day.

**Comment by DmitryAlexei** [comment]
@moonraver 
It is better to create new_rendering.xml (with a different name) and import it in OsmAnd (or open it using the OsmAnd app). Then choose your new rendering style in Configure map > map style
Changing the initial file may not always work.

**Reply by moonraver** [reply]
I don't know how to "propose a PR" in default rendering.xml and I don't want to brake something in your files. So I suggest some changes in a few rows in rendering.xml. I have tested these changes and now the differences among these colors are well visible in your maps. I would like to ask you if you could make this changes happen.  

**1. 3 colors - good, intermediate and bad smoothness has been changed:**
renderingConstant name="surfaceGradeGoodColor" value="#8bedff"/
renderingConstant name="surfaceGradeIntermediateColor" value="#c0f5ff"/
renderingConstant name="surfaceGradeBadColor" value="#ffc8c8"/


**2. Color of wood surface has been changed (was the same as stone, pebblestone):**
renderingAttribute name="surfaceWoodColor"
case attrColorValue="#ffd500"/

_I don't use NIGHT MODE but in this case "surfaceWoodColor" can be the same as DAY MODE (was also tested and was OK) or suggest any different color. So:_
case nightMode="true" attrColorValue="#ffd500"/


Thank you very much.
